### ISSUE TITLE
tried to install but got stuck

### ISSUE THREAD

**Issue by despiegk** [jgaskins/grpc#2]
```bash
JSX:OSX:grpc: shards install
Resolving dependencies
Fetching https://github.com/jgaskins/grpc.git
Fetching https://github.com/jeromegn/protobuf.cr.git
Installing grpc (0.1.0 at 6503b9b2a22f3a5cebf06c08f1319b7787c1386e)
Installing protobuf (2.2.0 at d7f31df6c72218ed8c038f1337c7ef105249c38e)
Postinstall of grpc: shards build && mkdir -p ../../bin && cp bin/grpc_crystal ../../bin
Failed postinstall of grpc on shards build && mkdir -p ../../bin && cp bin/grpc_crystal ../../bin:
Resolving dependencies
Fetching https://github.com/jeromegn/protobuf.cr.git
Using protobuf (2.2.0 at d7f31df6c72218ed8c038f1337c7ef105249c38e)
Unhandled exception: Error opening file with mode 'r': '/Users/despiegk/code/github/despiegk/crystal_tests/grpc/lib/grpc/lib/protobuf/bin/protoc-gen-crystal': No such file or directory (File::NotFoundError)
  from File::new<String, String, File::Permissions, Nil, Nil>:File
  from Shards::Package#install_executables:Nil
  from Shards::Commands::Install@Shards::Command::run<String>:Nil
  from ~procProc(Array(String), Array(String), Nil)@src/cli.cr:39
  from __crystal_main
  from main
```

my test repo is on https://github.com/despiegk/crystal_tests/tree/master/grpc

help greatly appreciated, would be amazing to use grpc in crystal

**Comment by jgaskins** [comment]
Hmm, looks like this might be a difference in how `shards` works in Crystal 0.34. Thanks for the heads up, looking into this now.

**Comment by jgaskins** [comment]
Yeah, looks like some changes to how post install scripts run in `shards`. `grpc` and `protobuf.cr` both build executables, but it's trying to run `postinstall` of `grpc` first. I'll file a bug in the `shards` repo.

**Comment by jgaskins** [comment]
That is, it's not running `postinstall` for dependencies before running it for the dependent shard, so the file it's expecting to be there isn't there. I have a feeling it might be because the dependency tree is flattened incorrectly before being iterated [here](https://github.com/crystal-lang/shards/blob/v0.10.0/src/commands/install.cr#L59-L70).

**Comment by jgaskins** [comment]
Filed a PR with the `shards` repo: https://github.com/crystal-lang/shards/pull/359

The downside is that even if that gets merged, it probably won't be released until Crystal 0.35. In the meantime, you can build your own `shards` binary from my branch and add it to your path.

**Comment by despiegk** [comment]
wow thank you, 
```
JSX:OSX:grpc: shards install
Resolving dependencies
Fetching https://github.com/jgaskins/grpc.git
Fetching https://github.com/jeromegn/protobuf.cr.git
Installing grpc (0.1.0 at 09c4947)
Installing protobuf (2.2.0 at d7f31df)
Postinstall of protobuf: shards build && mkdir -p ../../bin && cp bin/protoc-gen-crystal ../../bin
Postinstall of grpc: shards build
Writing shard.lock
```

worked for me

**Comment by jgaskins** [comment]
Looks like that isn't the fix they want because depending on how you set up your `shard.yml` the flattened dependency list could be in different orders. But a member of the Crystal core team [knows of a solution and has volunteered to implement it](https://github.com/crystal-lang/shards/pull/359#issuecomment-616012251).

In the meantime, this `shard.yml` will work with the released `shards` build on Crystal 0.34:

```yaml
dependencies:
  protobuf:
    github: jeromegn/protobuf.cr
    branch: master
  grpc:
    github: jgaskins/grpc
    branch: master
```

I also went ahead and removed the ugly `cp bin/...` hacks from the `postinstall` scripts of both repos now that the `executables` field in `shard.yml` is working. [It wasn't working before](https://github.com/crystal-lang/shards/issues/320), but apparently this new version fixed it.

**Comment by jgaskins** [comment]
This [bug is fixed in `shards`](https://github.com/crystal-lang/shards/pull/369) and will be in its next release. Until then, the workaround above will do the trick. :-) Thanks for reporting it!
